src/types/authenticator/code.ts

```diff
@@ -0,0 +1,182 @@
+import { URI } from 'vscode-uri';
+
+type Type = 'totp' | 'TOTP' | 'hotp' | 'HOTP';
+
+type AlgorithmType =
+    | 'sha1'
+    | 'SHA1'
+    | 'sha256'
+    | 'SHA256'
+    | 'sha512'
+    | 'SHA512';
+
+export class Code {
+    static readonly defaultDigits = 6;
+    static readonly defaultAlgo = 'sha1';
+    static readonly defaultPeriod = 30;
+
+    // id for the corresponding auth entity
+    id?: String;
+    account: string;
+    issuer: string;
+    digits?: number;
+    period: number;
+    secret: string;
+    algorithm: AlgorithmType;
+    type: Type;
+    rawData?: string;
+
+    constructor(
+        account: string,
+        issuer: string,
+        digits: number | undefined,
+        period: number,
+        secret: string,
+        algorithm: AlgorithmType,
+        type: Type,
+        rawData?: string,
+        id?: string
+    ) {
+        this.account = account;
+        this.issuer = issuer;
+        this.digits = digits;
+        this.period = period;
+        this.secret = secret;
+        this.algorithm = algorithm;
+        this.type = type;
+        this.rawData = rawData;
+        this.id = id;
+    }
+
+    static fromRawData(id: string, rawData: string): Code {
+        let santizedRawData = rawData
+            .replace(/\+/g, '%2B')
+            .replace(/:/g, '%3A')
+            .replaceAll('\r', '');
+        if (santizedRawData.startsWith('"')) {
+            santizedRawData = santizedRawData.substring(1);
+        }
+        if (santizedRawData.endsWith('"')) {
+            santizedRawData = santizedRawData.substring(
+                0,
+                santizedRawData.length - 1
+            );
```

Comment:
Shall we switch to this after I come back. Right now, have tested all functionality with current package. Switching it might break it and will require remote debugging with Vishnu